File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        Map<String, Integer> pokemon = new HashMap<>();
        Map<Integer, String> pokemonNum = new HashMap<>();

        for(int i=1; i<=n; i++) {
            String name = br.readLine();
            pokemon.put(name, i);
            pokemonNum.put(i, name);
        }

        StringBuilder sb = new StringBuilder();
        for(int i=0; i<m; i++) {
            String s = br.readLine();
            if(pokemon.containsKey(s))
                sb.append(pokemon.get(s) + "\n");
            else
                sb.append(pokemonNum.get(Integer.parseInt(s)) + "\n");
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.kt
fun main() = with(System.`in`.bufferedReader()) {
    val (n, m) = readLine().split(" ").map { it.toInt() }
    val sb = StringBuilder()
    val map = mutableMapOf<String, Int>()
    val arr = Array(n + 1) { "" }
    for (i in 1..n) {
        val name = readLine()
        map.put(name, i)
        arr[i] = name
    }
    repeat(m) {
        val query = readLine()
        if (query[0].isDigit()) {
            sb.appendLine(arr[query.toInt()])
        } else {
            sb.appendLine(map[query])
        }
    }
    print(sb)
}